app/src/utils/date-utils.js

```diff
@@ -45,3 +49,31 @@ export const convertNanosecondsToUTC = nanosecondTimestamp => {
     const offsetNS = new Date().getTimezoneOffset() * 60 * 10 ** 9;
     return nanosecondTimestamp + offsetNS;
 };
+
+/**
+ * @desc Given a nanosecond epoch date, returns time string in the format YYYY-MM-DD
+ * @param nanosecondDate
+ * @returns {string}
+ */
+export const epochToDateString = nanosecondDate => {
+    // We only need day precision, so get the date in milliseconds
+    const millisecondDate = nanosecondDate / NANOSECONDS_IN_ONE_MILLISECOND;
+    return moment.utc(millisecondDate).format('YYYY-MM-DD');
+};
+
+/**
+ * @desc Given a nanosecond epoch timestamp, returns an object with date nanoseconds and time nanoseconds
+ * @param nanosecondTimestamp
+ * @returns {{timeNanoseconds: Number, dateNanoseconds: Number}} The timestamp split into its date nanoseconds and
+ * its time nanoseconds
+ */
+export const splitNanosecondEpochTimestamp = nanosecondTimestamp => {
+    // eslint-disable-next-line no-param-reassign
+    if (typeof nanosecondTimestamp === 'string') nanosecondTimestamp = parseInt(nanosecondTimestamp);
```

Comment:
I think here you should actually fix it, (assign `nanosecondTimestamp` to a new variable and modify that variable)